src/Features/Core/Portable/FeaturesResources.resx

```diff
@@ -2948,4 +2948,12 @@ Zero-width positive lookbehind assertions are typically used at the beginning of
   <data name="Silent" xml:space="preserve">
     <value>Silent</value>
   </data>
+  <data name="embedded" xml:space="preserve">
+    <value>embedded</value>
+    <comment>Embedded is a technical term for "Embedded source", where souce files are embedded into the PDB</comment>
+  </data>
+  <data name="external" xml:space="preserve">
+    <value>external</value>
+    <comment>External means "external source", meaning source files that are not part of the current solution</comment>
```

Comment:
Do we really need to tag this specially, if it's opening a file on your machine? Like does this get strange if you then want to edit that file like a local file, or add that missing project to your solution?

Reply:
Or are we taking the on-disk file and copying it to a temp folder?

Reply:
No, you were right, this is a file on your machine.. but we open it readonly, and if you were to edit it, then we wouldn't find it because the checksum wouldn't match. If we didn't tag it specifically, what would we say? I don't think it should look like a normal file, because its not in the solution, so lots of things don't work like normal (and as mentioned, its readonly).  I don't think the tab should say "from metadata", because its the real source. So "external" is the best I could come up with. "local" I guess could work?